camelot/admin/action/form_action.py

```diff
@@ -255,11 +257,11 @@ def model_run( self, model_context ):
             if reply == QtWidgets.QMessageBox.Discard:
                 if admin.is_persistent( obj ):
                     admin.refresh( obj )
-                    yield action_steps.UpdateObjects((obj,))
+                    yield action_steps.UpdateObjects((subsystem_obj,))
                 else:
                     depending_objects = list(admin.get_depending_objects(obj))
-                    model_context.proxy.remove(obj)
-                    yield action_steps.DeleteObjects((obj,))
+                    model_context.proxy.remove(subsystem_obj)
```

Comment:
?

Reply:
Fixed in 3da5746ae8e4752ba43b94bebdae789c4438e6f0